samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -16,85 +16,327 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.apache.samza.operators.windows;
 
+import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.operators.data.MessageEnvelope;
+import org.apache.samza.operators.triggers.TimeTrigger;
+import org.apache.samza.operators.triggers.Trigger;
+import org.apache.samza.operators.triggers.Triggers;
 
+import java.time.Duration;
 import java.util.Collection;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
-
 /**
- * This class defines a collection of {@link Window} functions. The public classes and methods here are intended to be
- * used by the user (i.e. programmers) to create {@link Window} function directly.
+ * Factory methods for creating a {@link Window}. Provides APIs for specifying different types of {@link Window}s.
+ * A {@link Window} can have a specified keyFunction that is used to group {@link MessageEnvelope}s in the {@link Window}.
```

Comment:
Should be after the next section.  Instead of explaining what the keyFn is, this should explain what a keyed window means. keyFn should be described as part of the method docs.